trackintel/analysis/location_identification.py

```diff
@@ -143,33 +143,57 @@ def pre_filter_locations(
 
 def freq_recipe(sps, *labels):
     """Generate a location activity per user by assigning the most visited location the label "home"
-    and the second most visited location the label "work". The remaining location get no label.
+    and the second most visited location the label "work". The remaining locations get no label.
 
     Parameters
     ----------
     sps : GeoDataFrame (as trackintel staypoints)
         Staypoints with the column "location_id".
+    labels : collection of str, default ("home", "work")
+        Labels in decreasing time of activity.
 
     Returns
     -------
-    pd.Series
-        Staypoints with label given labels
+    GeoDataFrame
+        with column "activity_label".
 
     Examples
     --------
-    >> do something
+    >> staypoints["activity"] = ti.analysis.freq_recipe(staypoints, "home", "work")
     """
+    sps = sps.copy()
     if not labels:
         labels = ("home", "work")
-    if sps["duration"] not in sps.columns:
-        sps = sps.copy()
-        sps["duration"] = sps["finished_at"] - sps["started_at"]
-    return sps.groupby(["user_id"])["duration"].transform(_freq_transform, *labels)
+    for name, group in sps.groupby("user_id"):
+        if "duration" not in group.columns:
+            group["duration"] = group["finished_at"] - group["started_at"]
+        # pandas keeps inner order or groups
+        sps.loc[sps["user_id"] == name, "activity_label"] = _freq_transform(group, *labels)
+    return sps
 
 
-def _freq_transform(duration, *labels):
+def _freq_transform(group, *labels):
     """Transform function that assigns the longest visited data point the labels in order.
 
```

Comment:
possibly confusing statement: _assigns the longest visited data point the labels in order_

Reply:
Tried to improve the docstring.